// PB2BP: BP_FILE_NAME=PbGame
syntax = "proto3";
package idlepb;

import "defines.proto";
import "common.proto";

message Ping {
	int64 req_ticks = 1;
}
message Pong {
	int64 req_ticks = 1;
	int64 rsp_ticks = 2;
}

message DoGmCommand {
	string command = 1;
}

message ReportError {
	string text = 1;
}


enum LoginGameRetCode {
	LoginGameRetCode_Ok = 0;  // 正常登陆
	LoginGameRetCode_Unknown = 1;  // 未知错误
	LoginGameRetCode_NoRole = 2;  // 没有角色
	LoginGameRetCode_DuplicateLogin = 3;  // 已经在线
	LoginGameRetCode_AccountInvalid = 4;  // 帐号非法
	LoginGameRetCode_VersionError = 5;  // 版本错误
}

// 登录游戏	
message LoginGameReq {
	string account = 1;
	string client_version = 2;
}
message LoginGameAck {
	LoginGameRetCode ret = 1;
	PlayerData player_data = 2;
	bool is_relogin = 3;  // 是否为重连 (即服务器上玩家对象已经存在)
}
